controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/kvtable/CreateTableTask.java

```diff
@@ -0,0 +1,122 @@
+import io.pravega.client.tables.KeyValueTableConfiguration;
+import io.pravega.controller.store.stream.CreateStreamResponse;
+import io.pravega.controller.store.kvtable.KVTOperationContext;
+import io.pravega.controller.store.stream.State;
+import io.pravega.controller.stream.api.grpc.v1.Controller;
+import io.pravega.controller.task.KeyValueTable.TableMetadataTasks;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.controller.event.kvtable.CreateTableEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Request handler for performing scale operations received from requeststream.
+ */
+@Slf4j
+public class CreateTableTask implements TableTask<CreateTableEvent> {
+
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final TableMetadataTasks kvtMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public CreateTableTask(final KVTableMetadataStore kvtMetaStore,
+                           final TableMetadataTasks kvtMetaTasks,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(kvtMetaStore);
+        Preconditions.checkNotNull(kvtMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.kvtMetadataStore = kvtMetaStore;
+        this.kvtMetadataTasks = kvtMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final CreateTableEvent request) {
+        String scope = request.getScopeName();
+        String kvt = request.getKvtName();
+        int partitionCount = request.getPartitionCount();
+        long creationTime = request.getTimestamp();
+        long requestId = request.getRequestId();
+        String kvTableId = request.getTableId().toString();
+        KeyValueTableConfiguration config = new KeyValueTableConfiguration(partitionCount);
+
+        return kvtMetadataStore.getKVTable(scope, kvt, null).getId()
+        .thenCompose(id -> {
+            if (!id.equals(kvTableId)) {
+                // we don;t execute the request if Table IDs do not match
+                return CompletableFuture.completedFuture(null);
+            } else {
+                this.kvtMetadataStore.createKeyValueTable(scope, kvt, config, creationTime, null, executor)
+                        .thenComposeAsync(response -> {
+                            // only if its a new kvtable or an already existing non-active kvtable then we will create
+                            // segments and change the state of the kvtable to active.
+                            if (response.getStatus().equals(CreateKVTableResponse.CreateStatus.NEW) ||
+                                    response.getStatus().equals(CreateStreamResponse.CreateStatus.EXISTS_CREATING)) {
```

Comment:
change the CreateStreamResponse to CreateKVTResponse

Reply:
done